sweetest/src/main/java/com/mysugr/sweetest/framework/build/BaseBuilder.kt

```diff
@@ -100,6 +107,39 @@ abstract class BaseBuilder<TSelf, TResult : BaseAccessor>(
             createObject))
     }
 
+    // COROUTINES
+
+    /**
+     * [kotlinx.coroutines.Dispatchers.Main] can be set to the CoroutineScope provided by sweetest before each
+     * test and also reset it after. This behavior is **enabled by default**, but with this function it can be forced to
+     * be disabled or enabled for a test.
+     *
+     * If there are conflicting calls to this function, an exception is thrown (e.g. in the test `true` is passed, but
+     * in one of the steps `false` is passed).
+     */
+    fun autoSetMainCoroutineDispatcher(value: Boolean) = apply {
```

Comment:
Just an idea: Since only `testContext` is used here (which is effectively public), this could be an extension function on `BaseBuilder`.
The advantage would be that all the Coroutine Code could reside together, all the Mocking Code could reside together, etc. In the long run, we could extract those parts and could provide _sweetest_ with a _coroutines_-extension, a _mocking_-extension,... At the bottom line: modularity.

Reply:
that's a good point actually. on the other side i planned on ditching the idea to have extensions for these kinds of tasks so people can just control + click members and immediately what the classes are capable of. that would imply e.g. `invoke` operator, `steps` and `dependencies` which i find inconvenient to having it to import all the time. also it makes the usage less self-explaining. maybe it's a good common ground to have some core features and others be considered extensions (as that's what extension features are here for)?

Reply:
> maybe it's a good common ground to have some core features and others (custom ones) be considered extensions...

Valid approach. It's just hard to decide where to draw the line then.
I think it's fine for now. I just wanted to point it out so it's not forgotten.